teps/0143-concise-parameters-results.md

```diff
@@ -0,0 +1,719 @@
+    description: Name of the Cloud Function.
+    required: true
+  description:
+    description: Description for the Cloud Function.
+    required: false
+  runtime:
+    description: Runtime to use for the function.
+    required: true
+outputs:
+  url:
+    description: The URL of your Cloud Function. Only available with HTTP Trigger.
+  id:
+    description: Full resource name of the Cloud Function, of the format 'projects/p/locations/l/functions/f'.
+  status:
+    description: Status of the Cloud Function deployment.
+  version:
+    description: Version of the Cloud Function deployment.
+  runtime:
+    description: Runtime of the Cloud Function deployment.
+runs:
+  using: 'node16'
+  main: 'dist/index.js'
+```
+
+## Proposal
+
+We propose using maps for `Params` and `Results` instead of a list of sub-objects containing names.
+
+### Syntax
+
+#### Parameters
+
+At authoring time, users declare string, array and object `Parameters` for `Tasks` and `Pipelines`. It is tedious to specify these fields because they are defined as sub-objects. We propose using maps instead of sub-objects to declare `Parameters`. The new syntax of `Params` will be first added to `v2alpha1`
+
+```yaml
+# before in v1
+params:
+  - name: hello
+    type: string
+  - name: environments
+    type: array
+  - name: git
+    type: object
+    properties:
+      url: {}
+      commit: {}
+
+# after in v2
+params:
+  hello:
+    type: string
+  environments:
+    type: array
+  git:
+    type: object
+    properties:
+      url: {}
+      commit: {}
+```
+
+At runtime, users configure execution by passing in `Parameters`. We propose using maps instead of sub-objects to declare `Parameters` at runtime.
+
+```yaml
+# before in v1
+params:
+  - name: hello
+    value: world
+  - name: environments
+    value:
+      - 'staging'
+      - 'qa'
+      - 'prod'
+  - name: git
+    value:
+      url: abc.com
+      commit: sha123
+
+# after in v2
+params:
+  hello: world
+  environments:
+    - 'staging'
+    - 'qa'
+    - 'prod'
+  git:
+    url: abc.com
+    commit: sha123
+```
+
+#### Results
+
+At authoring time, users declare string, array and object `Results` for `Tasks` and `Pipelines`. It is tedious to specify these fields because they are defined as sub-objects. We propose using maps instead of sub-objects to declare `Results`.
+
+```yaml
+# before in v1
+results:
+  - name: hello
+    type: string
+  - name: environments
+    type: array
+  - name: git
+    type: object
+    properties:
+      url: {}
+      commit: {}
+
+# after in v2
+results:
+  hello:
+    type: string
+  environments:
+    type: array
+  git:
+    type: object
+    properties:
+      url: {}
+      commit: {}
+```
+
+At runtime, `Results` are stored in the status of `TaskRuns`, `PipelineRuns` and `CustomRuns` as sub-objects. We propose using maps instead of sub-objects to produce `Results` at runtime. We also propose removing the type field that’s added by default to all `Results` in status.
```

Comment:
About the removal of `type`, I guess that's because it can be inferred from the format of the result?